sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_connection_async.py

```diff
@@ -439,7 +444,7 @@ async def _incoming_open(self, channel, frame):
             _LOGGER.error("OPEN frame received in the OPENED state.", extra=self._network_trace_params)
             await self.close()
         if frame[4]:
-            self._remote_idle_timeout = frame[4] / 1000  # Convert to seconds
+            self._remote_idle_timeout = cast(float, frame[4] / 1000)  # Convert to seconds
             self._remote_idle_timeout_send_frame = (
```

Comment:
does frame[4] need to get deserialized as an int?

Reply:
i.e could it be a float and save us the cast

Reply:
the issue is the line below where we do the multiplication. mypy isn't seeing that `_remote_idle_timeout` is now a `float` after the division, hence the cast.